packages/jsii-pacmak/lib/targets/python.ts

```diff
@@ -630,103 +607,159 @@ class InterfaceProperty extends BaseProperty {
-                member.emit(code, resolver);
-            }
-            code.closeBlock();
+        code.closeBlock();
+    }
 
-            // Now we'll emit the mandatory members.
-            code.line(`@jsii.data_type(jsii_type="${this.fqn}", jsii_struct_bases=[_${this.name}])`);
-            code.openBlock(`class ${this.name}(_${this.name})`);
-            emitDocString(code, this.docs);
-            for (const [member, sep] of separate(sortMembers(mandatoryMembers, resolver))) {
-                if (sep) { code.line(''); }
-                member.emit(code, resolver);
-            }
-            code.closeBlock();
-        } else {
-            code.line(`@jsii.data_type(jsii_type="${this.fqn}", jsii_struct_bases=[${baseInterfaces.join(', ')}])`);
+    protected getClassParams(resolver: TypeResolver): string[] {
+        return this.bases.map(b => resolver.resolve({ type: b }));
+    }
 
-            // In this case we either have no members, or we have all of one type, so
-            // we'll see if we have any optional members, if we don't then we'll use
-            // total=True instead of total=False for the class.
-            if (optionalMembers.length >= 1) {
-                code.openBlock(`class ${this.name}(${classParams.concat(["total=False"]).join(", ")})`);
-            } else {
-                code.openBlock(`class ${this.name}(${classParams.join(", ")})`);
-            }
-            emitDocString(code, this.docs);
+    /**
+     * Find all fields (inherited as well)
+     */
+    private get allMembers(): StructField[] {
+        return this.thisInterface.allProperties.map(x => new StructField(propertySpec(x)));
+    }
 
-            // Finally we'll just iterate over and emit all of our members.
-            if (this.members.length > 0) {
-                for (const [member, sep] of separate(sortMembers(this.members, resolver))) {
-                    if (sep) { code.line(''); }
-                    member.emit(code, resolver);
-                }
-            } else {
-                code.line("pass");
-            }
+    private get thisInterface() {
+        if (this.fqn === null) { throw new Error('FQN not set'); }
+        return this.generator.reflectAssembly.system.findInterface(this.fqn);
+    }
 
-            code.closeBlock();
+    private emitConstructor(code: CodeMaker, resolver: TypeResolver) {
+        const members = this.allMembers;
+
+        const kwargs = members.map(m => m.constructorDecl(resolver));
+
+        const constructorArguments = kwargs.length > 0  ? ['self', '*', ...kwargs] : ['self'];
+
+        code.openBlock(`def __init__(${constructorArguments.join(', ')})`);
+        this.emitConstructorDocstring(code);
+
+        // Required properties, those will always be put into the dict
+        code.line('self._values = {');
+        for (const member of members.filter(m => !m.optional)) {
+            code.line(`    '${member.pythonName}': ${member.pythonName},`);
```

Comment:
Do we need to validate that we actually got values in all required fields or are we leaving that to jsii?

Reply:
Fair point. It's easy enough to add, might as well do it.

Reply:
Ah no, we don't have to, Python will complain. The constructor looks like this:

```ts
def __init__(self, *, required, optional=None):
  ...
```

So you MUST pass a value for required.